lib/logitech_receiver/base.py

```diff
@@ -189,6 +201,17 @@ def read(handle, timeout=DEFAULT_TIMEOUT):
 		return reply[1:]
 
 
+# sanity checks on  message report id and size
+def check_message(data) :
+	assert isinstance(data, bytes), (repr(data), type(data))
+	report_id = ord(data[:1])
+	if report_id <= max_nonDJ_report_id or report_lengths.get(report_id) == len(data):
```

Comment:
Can you clarify the deal with `max_nonDJ_report_id`?

Reply:
The code used to test report_id & 0xF0 == 0x00 to determine whether the message was an old-style device report.  I changed this to check report_id <= max_nonDJ_report_id instead in an attempt to make the code somewhat clearer.

Reply:
You mean like `0x10`, `0x11` and `0x12`? If so please drop the constant and check each report individually. It is not clear to where that magic constant comes from.

Reply:
This is for device reports with report ids like 0x02 (mouse movement) from receivers that don't use the DJ reporting.   
I have two different receiver/device setups:

- a c52b unifying receiver and an M510 mouse. This mouse/receiver produces DJ movement reports that this PR with -dd flag reports as 
04:43:54,020    DEBUG [ReceiverListener:hidraw5] logitech_receiver.base: (14) => r[20 01 0200 00FD3F0000000000000000]
Without this PR Solaar turns these device reports into Solaar notifications and then complains about them as in
04:47:35,844    DEBUG [ReceiverListener:hidraw5] logitech_receiver.base: (14) => r[20 01 0200 0001E0FF00000000000000]
04:47:35,844    DEBUG [ReceiverListener:hidraw5] logitech_receiver.notifications: <PairedDevice(1,1025,M510,93E8527F)> (1.0) custom notification Notification(1,02,00,0001E0FF00000000000000)
04:47:35,844  WARNING [ReceiverListener:hidraw5] logitech_receiver.notifications: <PairedDevice(1,1025,M510,93E8527F)>: unrecognized Notification(1,02,00,0001E0FF00000000000000)
A major point of this PR is to remove this incorrect handling of such device reports.

- a c534 (nano) receiver and an M185 mouse - this is from the low-end MK270 mouse/keyboard product.  This mouse produces movement reports like:
04:42:27,502    DEBUG [ReceiverListener:hidraw2] logitech_receiver.base: (15) => r[02 00000890FF0000] ignoring unknown report
They were allowed and then filtered using the & 0xF0 code, which I replaced with the check against the symbolic maximum in an attempt to improve code readability.  Maybe this symbolic constant should have a different name?  (But I'm not sure that else to call it.)

Reply:
This behavior is not correct. The report IDs for non HID++ actions, such as mouse movement, are defined in the report descriptor of the HID device and they could be whatever Logitech wants. The correct approach would be parsing the report descriptor and seeing if the report ID is from a vendor usage page, I am doing this in logitechd. However, as this requires an ioctl just to get the report descriptor from the kernel which we would then have to parse manually, I am fine with just checking if the report ID is one of the known HID++/DJ report IDs.

So, please replace `report_id <= max_nonDJ_report_id` with `report_id in report_lenghts`.

Reply:
The low-level code for reading messages is a bit of a mess in Solaar, but here is what I see with -dd (with this PR).

From my unifying receiver I see report ids 10, 11, and 20, as in:

05:28:11,498    DEBUG [MainThread] logitech_receiver.base: (14) => r[11 FF 83B5 031EB692A30106120000000000000000]
05:28:11,599    DEBUG [MainThread] logitech_receiver.base: (14) => r[20 00 4102 0000000000000000000000]
05:28:11,599    DEBUG [MainThread] logitech_receiver.base: (14) => r[20 01 4101 2510040000000000000000]
05:28:11,599    DEBUG [MainThread] logitech_receiver.base: (14) => r[20 02 4101 11201A4000000000000000]
05:28:11,600    DEBUG [MainThread] logitech_receiver.base: (14) => r[10 FF 8F80 B20300]

Part of this pull request is to better handle the messages with report id 20.  Previously Solaar created Solaar notifications from them which then caused warning messages.


From my non-unifying receiver I see report ids 10, 11, and 02, as in

05:32:02,857    DEBUG [ReceiverListener:hidraw2] logitech_receiver.base: (14) <= w[10 02 000A 000100]
05:32:01,088    DEBUG [ReceiverListener:hidraw2] logitech_receiver.base: (14) => r[11 FF 83B5 316FAD17D5040000000F000000000000]
05:32:02,852    DEBUG [ReceiverListener:hidraw2] logitech_receiver.base: (14) => r[02 0000F91F000000] ignoring unknown report

Not so long ago I put in PR #623, which turned off the logging of the messages with report id under 0x10 because I was seeing a lot of them.  I think this was a mistake so part of this PR turns on this logging again.   A big reason to again log these messages is because the DJ messages (0x20) are also logged.

The code to detect these messages is (report_id & 0xF0 == 0) which has a magic hard constant in it. Also messages in question are those with report id less than 0x10, which is better checked arithmetically.  I changed the check and used the symbolic constant max_nonDJ_report_id = 0x0F so that the check is now report_id <= max_nonDJ_report_id.


It sounds like you are asking to log and ignore any message with report id that is not 10, 11, 20, or 21.  I'm fine with that, but please verify.

Reply:
> It sounds like you are asking to log and ignore any message with report id that is not 10, 11, 20, or 21. I'm fine with that, but please verify.

Yes.

Reply:
Well, not log. Just ignore.

Reply:
OK, I left in the debug log code for these messages, commented out, so that it can be added back if needed.